python/pyspark/pandas/tests/indexes/test_base.py

```diff
@@ -17,7 +17,7 @@
 
 import inspect
 import unittest
```

Comment:
It's my phone so please ignore this comment if it's wrong. If we use distutils only for LooseVersion, we could just have our own LooseVersion in PySpark too (instead of embedding the package)

Reply:
It also sounds good if `LooseVersion` is easy to port into Spark.

Reply:
It's not difficult to copy or to reimplement, @HyukjinKwon and @viirya . 

However, it's a PSF License which is different from Py4J or cloudpickle (under BSD 3-Clause), 
https://github.com/apache/spark/blob/58c24a5719b8717ea37347c668c9df8a3714ae3c/LICENSE-binary#L460-L462

I believe it's compatible but we need to take a look once more before doing that. It's because our Apache Spark binary distribution doesn't include `Python Software Foundation` yet.

Reply:
We could just write our own instead of copying. It's just a version string check in the end. I have seen their code yet so I can write one on my own too :-).